persistent/cPersistence.c

```diff
@@ -917,21 +921,73 @@ unghost_getattr(const char *s)
     return 1;
 }
 
+/* Returns pointer of attr if attr is exempt from ghostification because it has been annotated.
+
+   Some attributes don't need unghostification because they are never going to be modified.
+   They are faster to access with the tradeoff of higher memory usage and immutability.
+   They are always read from the class (not instance) of the Persistent.
+   This facility is designed for non-threaded implementations of the underlying storage,
+   so that they can avoid being called.
+   Especially, asyncio implementations can avoid unnecessary `await` operators
+*/
+static PyObject *
+is_static(cPersistentObject *self, PyObject *attr)
+{
+    PyObject *clattr = PyObject_GenericGetAttr((PyObject *)Py_TYPE(self), attr);
+    if (clattr == NULL)
+    {
+        PyErr_Clear();
+        return NULL;
+    }
+    if (PyObject_IsInstance(clattr, (PyObject *)&Statictype))
+    {
+        return clattr;
+    }
+    else
+    {
+        Py_DECREF(clattr);
+        return NULL;
+    }
+}
+
+static PyObject *
+get_static(cPersistentObject *self, PyObject *attr)
+{
+    PyObject *clattr;
+    clattr = 0;
+    clattr = is_static(self, attr);
+    PyObject *value;
```

Comment:
It looks like the version of MSVC++ used to build persistent for Python 2.7 on Windows in Appveyr doesn't support C99 variable declarations in the middle of a block:

```
    C:\Users\appveyor\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\cl.exe /c /nologo /Ox /MD /W3 /GS- /DNDEBUG -Ic:\python27\include -Ic:\python27\PC /Tcpersistent/cPersistence.c /Fobuild\temp.win32-2.7\Release\persistent/cPersistence.obj
    cPersistence.c
    persistent/cPersistence.c(959) : error C2275: 'PyObject' : illegal use of this type as an expression
            c:\python27\include\object.h(108) : see declaration of 'PyObject'
    persistent/cPersistence.c(959) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(962) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(962) : warning C4047: '=' : 'int' differs in levels of indirection from 'PyObject *'
    persistent/cPersistence.c(963) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(966) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(966) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(966) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(966) : warning C4047: 'function' : 'PyObject *' differs in levels of indirection from 'int'
    persistent/cPersistence.c(966) : warning C4024: 'function through pointer' : different types for formal and actual parameter 1
    persistent/cPersistence.c(966) : warning C4133: 'function' : incompatible types - from 'cPersistentObject *' to 'PyObject *'
    persistent/cPersistence.c(966) : warning C4047: '=' : 'int' differs in levels of indirection from 'PyObject *'
    persistent/cPersistence.c(968) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(970) : error C2065: 'value' : undeclared identifier
    persistent/cPersistence.c(970) : warning C4047: 'return' : 'PyObject *' differs in levels of indirection from 'int'
    error: command 'C:\\Users\\appveyor\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\cl.exe' failed with exit status 2
```